core/primitives/src/sharding.rs

```diff
@@ -164,6 +167,26 @@ pub struct ShardChunk {
     pub header: ShardChunkHeader,
     pub transactions: Vec<SignedTransaction>,
     pub receipts: Vec<Receipt>,
+    /// Honeypot for catching lazy block producers
+    pub honeypot: Vec<Color>,
+}
+
+#[derive(
```

Comment:
This needs a rustdoc. Also, why not bool?

Reply:
I prefer to be explicit. http://blakesmith.me/2019/05/07/rust-patterns-enums-instead-of-booleans.html

Reply:
I what to thank @bowenwang1996 for embracing enums instead of bools as it makes the code more readable!

Should we probably call the structure `HoneypotColor`? `Color` term is too generic. Also, `red` and `blue` do not really help to understand what they imply. Are we sure we want to use colors?

Reply:
Unless the color is the part of the algorithm spec we shouldn't be using it.

Reply:
The colors is how it is written in the paper.